### PR TITLE
fix(png): fix mem leak in libpng decoder

### PR DESCRIPTION
**Pull request by lion2tomato**
fix mem leak in libpng decoder

### Notes
- Update the [Documentation](https://github.com/lvgl/lvgl/tree/master/docs) if needed.
- Add [Examples](https://github.com/lvgl/lvgl/tree/master/examples) if relevant.
- Add [Tests](https://github.com/lvgl/lvgl/blob/master/tests/README.md) if applicable.
- If you added new options to `lv_conf_template.h` run [lv_conf_internal_gen.py](https://github.com/lvgl/lvgl/blob/master/scripts/lv_conf_internal_gen.py) and update [Kconfig](https://github.com/lvgl/lvgl/blob/master/Kconfig).
- Run `scripts/code-format.py` (`astyle v3.4.12` needs to installed by running `cd scripts; ./install_astyle.sh`) and follow the [Code Conventions](https://docs.lvgl.io/master/CODING_STYLE.html).
- Mark the Pull request as [Draft](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request) while you are working on the first version, and mark is as _Ready_ when it's ready for review.
- When changes were requested, [re-request review](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review) to notify the maintainers.
- Help us to review this Pull Request! Anyone can [approve or request changes](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/approving-a-pull-request-with-required-reviews).

### CODE REVIEW

**Review by FASTSHIFT — APPROVED** [review]

**Review by FASTSHIFT — CHANGES_REQUESTED** [review]
Please remove `lazygit` and `lazygit.tar.gz` files.

**Review by kissa96 — APPROVED** [review]
LGTM

**Comment by kisvegabor** [comment]
> Please remove lazygit and lazygit.tar.gz files.

Ok for me too after removing `lazygit` and `lazygit.tar.gz`.

**Comment by lion2tomato** [comment]
removed lazygit and lazygit.tar.gz

**Comment by lion2tomato** [comment]
> Please remove `lazygit` and `lazygit.tar.gz` files.

have removed lazygit` and `lazygit.tar.gz` files.

**Review by FASTSHIFT — APPROVED** [review]

File: src/libs/libpng/lv_libpng.c [hunk 1]
```diff
@@ -303,6 +303,7 @@ static lv_draw_buf_t * decode_png(lv_image_decoder_dsc_t * dsc)
         if(dsc->src_type == LV_IMAGE_SRC_FILE) {
             LV_LOG_ERROR("alloc PNG_IMAGE_SIZE(%" LV_PRIu32 ") failed: %s", (uint32_t)PNG_IMAGE_SIZE(image),
                          (const char *)dsc->src);
+            png_image_free(&image);
```
Comment (XuNeo) [inline src/libs/libpng/lv_libpng.c:306]: Should we move logic to just before return NULL:
```
png_image_free(&image);
return NULL;
```

Comment (FASTSHIFT) [reply src/libs/libpng/lv_libpng.c:306]: Currently in the error handling branch, no need to return NULL in advance.

Comment (kisvegabor) [reply src/libs/libpng/lv_libpng.c:306]: I think @XuNeo means we should handle this case too:
image

which makes sense to me.

BTW, where `png_data` is freed?

Comment (XuNeo) [reply src/libs/libpng/lv_libpng.c:306]: `png_data`  is free'd in line 306, which is only necessary for file type of image.

**Review by kisvegabor — APPROVED** [review]

**Pull request merged**